zitilib/actions/edge/ctrl_add_routers.go

```diff
@@ -0,0 +1,129 @@
+package edge
+
+import (
+	"errors"
+	"fmt"
+	"github.com/Jeffail/gabs"
+	"github.com/netfoundry/fablab/kernel/model"
+	"github.com/netfoundry/fablab/zitilib/actions/cli"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+func CtrlAddRouters(regionSpec, hostSpec, componentSpec string) model.Action {
+	return &ctrlAddRouters{
+		regionSpec:    regionSpec,
+		hostSpec:      hostSpec,
+		componentSpec: componentSpec,
+	}
+}
+
+func (ar *ctrlAddRouters) Execute(m *model.Model) error {
+
+	hosts := m.GetHosts(ar.regionSpec, ar.hostSpec)
+	for _, h := range hosts {
+		components := h.GetComponents(ar.componentSpec)
+		for _, c := range components {
+			if c.Data == nil {
+				c.Data = model.Data{}
+			}
+
+			existingRouter, err := ar.getRouter(m, c)
+
+			if err != nil {
+				return err
+			}
+
+			if existingRouter.Data() == nil {
+				existingRouter, err = ar.createRouter(m, c)
+				if err != nil {
+					return err
+				}
+			}
+
+			if existingRouter.Path("enrollmentJwt").Data() == nil {
+				c.Data["isEnrolled"] = true
```

Comment:
I wrote this but it seems like the db is wiped after every `fablab up`. So I haven't had a chance to see it run.